### DISCUSSION TITLE
Writing encrypted PDFs? (or warning that it doesn't?)

### DISCUSSION THREAD

**Discussion by KevinDCanada** [pdfarranger/pdfarranger#1075] [Q&A]
I used PDF Arranger on Ubuntu/Mint and was wowed by the fantastic interface and number of features it incorporates into it (thank you!) so when I found it had a Windows port I was excited to try to get it approved for my organization.

Unfortunately what would seem to impede any enterprise usage is that although it handles importing encrypted PDFs, it doesn't seem to write them... there's no option in Save As or elsewhere I could find. As well, it doesn't seem to warn that the encryption is gone and the processed PDF will be saved unencrypted. For non-technical people in a security-conscious enterprise, this could be an issue.

Before I  submit this as a lightbulb idea, I just wanted to ensure that I wasn't missing anything and indeed PDF Arranger can't write encrypted PDFs nor warn that unencrypted ones will be written. Thanks!

**Comment by dreua** [comment]
Would a warning during decryption be enough?

**Reply by kbengs** [reply]
> it doesn't seem to write them

That is true.

For the coming release there is however some support:
If a encrypted (password protected) pdf is opened and then saved, then the file will be saved with the same password. The file with the password need to be added to PDF Arranger as the first file for this to work.

There is no way to change or set a new password. But I think it is something we could add in the future.

> Would a warning during decryption be enough?

Or the warning could be in the "Note the limitations" window at start. Or in the linked "User manual".

**Reply by KevinDCanada** [reply]
Either one of those would be excellent, especially the latter. I would suggest that for the latter it be noted somewhere in the app. possibly on the password dialog that if it's the first opened PDF it will be written encrypted with the same password and if not then it will be written with no excryption. As long as it informs, then it alleviates any security concerns.

Thank you both for the replies!
